### PR TITLE
[Fixes #213] Fix tpop buf

### LINKED ISSUE #213: Total population endpoint is not correct
- Kansas population numbers are suspect
- The real ideal pop per district is 734,470 based on 2020 census, but the module shows ideal pop of 531,545

In the `total_pop_udf` it looks like it gets either the VAP or TOTPOP. Which is correct?

https://github.com/districtr/districtr-v2/blob/bf57b0dce62e8d3c55e17e5e54674f843fc5d831/backend/app/sql/total_pop_udf.sql#L22

### PR DESCRIPTION
**Pull request by raphaellaude**
## Description

- Fixes #213 

## Reviewers

- Primary: @nofurtherinformation @mariogiampieri 
- Secondary:

## Checklist

- [ ] Added/Updated related documentation (if applicable).
- [x] Added/Updated related unit tests (if applicable).

## Screenshots (if applicable):

See with e.g. RI P4 view
image

### CODE REVIEW

File: app/src/app/components/sidebar/charts/HorizontalBarChart.tsx [hunk 1]
```diff
@@ -86,6 +86,9 @@ export const HorizontalBar = () => {
   }
 
   if (mapMetrics?.isError) {
+    if (mapMetrics?.error.response.data.detail) {
+      return <div>Error: {mapMetrics?.error.response.data.detail}</div>;
```
Comment (nofurtherinformation) [inline app/src/app/components/sidebar/charts/HorizontalBarChart.tsx:90]: PP: Remove extra conditional and return and have inline option chaining, like:

```javascript
<div>
  Error: {mapMetrics?.error.response.data.detail || mapMetrics?.error?.message}
</div>
```

File: app/src/app/store/mapStore.ts [hunk 2]
```diff
@@ -52,6 +53,12 @@ const combineSetValues = (setRecord: Record<string, Set<unknown>>, keys?: string
 const prodWrapper: typeof devtools = (store: any) => store;
 const devwrapper = process.env.NODE_ENV === 'development' ? devtools : prodWrapper;
 
+interface APIErrorResponse extends AxiosResponse {detail: string}
+
+export interface APIError extends AxiosError {
+  response: APIErrorResponse
```
Comment (nofurtherinformation) [inline app/src/app/store/mapStore.ts:59]: Nice

**Review by nofurtherinformation — CHANGES_REQUESTED** [review]
One picky minor thing in the JSX but otherwise LGTM!

**Comment by raphaellaude** [comment]
@nofurtherinformation this is finally ready for review! Working as expected in the test app :)

**Review by nofurtherinformation — APPROVED** [review]
lgtm! (may need one more mergeroni)

**Pull request merged**